### DISCUSSION TITLE
Input motor power commands directly in the client.

### DISCUSSION THREAD

**Discussion by pmaggs2** [bitcraze/discussions#1058] [Q&A]
Is there anyway i could input motor power commands in the client similar to how the firmware uses "powerSet(0, 0, 0, 0);" thanks

**Comment by knmcguire** [comment]
Yes that is possible. Have you found this parameter group as well?

https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#motorpowerset

Not sure how this would work for real time flight though if you are using an external controller. We have heard mixed results from that so you might want to consider sending pitch roll yaw instead

**Reply by NageshEranki** [reply]
Hi

I have tried naively setting the motor power commands (in a 'while' loop) through the parameter framework in my client (using an external controller). The drone cannot achieve stable hover this way. Is this because param updates are sent over radio at a much slower rate than perhaps a setpoint?

Thanks

**Reply by knmcguire** [reply]
yeah. that is known for being a lot slower... If you don't want to change anything in the firmware, the 'lowest' you can go for the 4xcascaded pid controller, is to do 'reverse motormixing' offboard ( to get roll pitch yaw rates and thrust from your controller), send it with send_setpoint from the [cflib commander class](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/commander/) while setting the [flight modes](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#flightmode) for roll and pitch to rate (yaw is rate by standard).

If you want to change the firmware, I've currently made an issue for this that hopefully reminds us to add it in the future. There are some difficulties with implementing it, like stopping the controller to not overwrite the values, but it should be possible. I've added some posts from the old forum that might be helpful: https://github.com/bitcraze/crazyflie-firmware/issues/1332

**Reply by knmcguire** [reply]
Also this discussion is interesting: https://github.com/orgs/bitcraze/discussions/184
